test/Altinn.Platform.Events.Tests/TestingControllers/EventsControllerTests.cs

```diff
@@ -123,6 +128,212 @@ public async Task Post_ExternalEventsDisabled_NotFoundResponse()
+                ProblemDetails actual = JsonSerializer.Deserialize<ProblemDetails>(content, _options);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+                Assert.Equal(expected, actual.Detail);
+            }
+
+            /// <summary>
+            /// Scenario:
+            ///   Retrieve a list of events, without bearer token.
+            /// Expected result:
+            ///   Returns HttpStatus Unauthorized.
+            /// Success criteria:
+            ///   The response has correct status.
+            /// </summary>
+            [Fact]
+            public async void GetEvents_MissingBearerToken_ReturnsUnauthorized()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}/events?after=0&subject=%2Fparty%2F1337";
+                HttpClient client = GetTestClient(new Mock<IEventsService>().Object);
+
+                HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri);
+
+                // Act
+                HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            }
+
+            /// <summary>
+            /// Scenario:
+            ///   Get events with  a valid set of query parameters
+            /// Expected result:
+            ///   Returns a list of events and a next header
+            /// Success criteria:
+            ///   The response has correct count. Next header is correct.
+            /// </summary>
+            [Fact]
+            public async void GetEvents_ValidRequest_ReturnsListOfEventsAndNextUrl()
+            {
+                // Arrange
+                string requestUri = $"{BasePath}/events?source=urn:altinn:systemx&after=0&subject=%2Fparty%2F1337";
+                string expectedNext = $"https://platform.localhost:5080/events/api/v1/events?after=e31dbb11-2208-4dda-a549-92a0db8c8808&source=urn:altinn:systemx&subject=/party/1337";
+                int expectedCount = 2;
+
+                HttpClient client = GetTestClient(new EventsServiceMock(3));
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetOrgToken("digdir", scope: "altinn:events.subscribe"));
+
+                HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri);
+
+                // Act
+                HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+                string responseString = await response.Content.ReadAsStringAsync();
+                List<CloudEvent> actual = JsonSerializer.Deserialize<List<CloudEvent>>(responseString);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                Assert.Equal(expectedCount, actual.Count);
```

Comment:
Just a heads up. The Count is indirectly produced by a mock. Not by the code under test.

Reply:
Good point, no sense in cluttering our assert list.